pkg/cloud/ovirt/machine/actuator.go

```diff
@@ -376,7 +377,85 @@ func (actuator *OvirtActuator) reconcileConditions(
 	return conditions
 }
 
+// validateMachine validates the machine object yaml fields and
+// returns InvalidMachineConfiguration in case the validation failed
 func (actuator *OvirtActuator) validateMachine(machine *machinev1.Machine, config *ovirtconfigv1.OvirtMachineProviderSpec) *apierrors.MachineError {
+
+	// UserDataSecret
+	if config.UserDataSecret == nil {
+		return apierrors.InvalidMachineConfiguration(
+			fmt.Sprintf("%s UserDataSecret must be provided!", ErrorInvalidMachineObject))
+	} else if config.UserDataSecret.Name == "" {
+		return apierrors.InvalidMachineConfiguration(
+			fmt.Sprintf("%s UserDataSecret *Name* must be provided!", ErrorInvalidMachineObject))
+	}
+
+	err := validateInstanceID(config)
+	if err != nil {
+		return err
+	}
+
+	// CredentialsSecret
+	if config.CredentialsSecret == nil {
+		return apierrors.InvalidMachineConfiguration(
+			fmt.Sprintf("%s CredentialsSecret must be provided!", ErrorInvalidMachineObject))
+	} else if config.CredentialsSecret.Name == "" {
+		return apierrors.InvalidMachineConfiguration(
+			fmt.Sprintf("%s CredentialsSecret *Name* must be provided!", ErrorInvalidMachineObject))
+	}
+
+	// root disk of the node
+	if config.OSDisk == nil {
+		return apierrors.InvalidMachineConfiguration(
+			fmt.Sprintf("%s OS Disk (os_disk) must be specified!", ErrorInvalidMachineObject))
+	}
+
+	return validateVirtualMachineType(config.VMType)
+}
+
+// validateInstanceID execute validations regarding the InstanceID.
+// Returns: nil or InvalidMachineConfiguration
+func validateInstanceID(config *ovirtconfigv1.OvirtMachineProviderSpec) *apierrors.MachineError {
+	// Cannot set InstanceTypeID and at same time: MemoryMB OR CPU
+	if len(config.InstanceTypeId) != 0 {
+		if config.MemoryMB != 0 || config.CPU != nil {
+			return apierrors.InvalidMachineConfiguration(
+				fmt.Sprintf("%s InstanceTypeID and MemoryMB OR CPU cannot be set at the same time!", ErrorInvalidMachineObject))
+		}
+	}
+
+	// If InstanceTypeId is NOT SET users can set MEM and CPU,
+	// all memory and cpu variables will be ignored.
+	if len(config.InstanceTypeId) == 0 {
```

Comment:
This "if" is redundant, please remove it.
Plus I think the comment is wrong/not clear and not needed :)